File: app/src/main/java/com/syncplay/ListDevicesActivity.java
package com.syncplay;

import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.widget.ListView;

public class ListDevicesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_devices);
        ((ListView)findViewById(R.id.devicesList)).setAdapter(new DeviceListAdapter(this));
        Toolbar toolbar = (Toolbar) findViewById(R.id.listToolbar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.list_devices_menu, menu);
        return true;
    }
}

File: app/src/main/java/com/syncplay/ListDevicesActivity.kt
package com.syncplay

import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.Menu

class ListDevicesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_devices)
        val devicesList = findViewById(R.id.devicesList) as RecyclerView
        devicesList.adapter = DeviceListAdapter(this)
        devicesList.layoutManager = LinearLayoutManager(this)
        val toolbar = findViewById(R.id.listToolbar) as Toolbar
        setSupportActionBar(toolbar)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.list_devices_menu, menu)
        return true
    }
}
